src/lib.rs

```diff
@@ -105,14 +106,13 @@ pub fn execvp<S, I>(program: S, args: I) -> Error
         CString::new(arg.as_ref().as_bytes())
     }).collect::<Result<Vec<_>, _>>());
     let mut arg_charptrs: Vec<_> = arg_cstrings.iter().map(|arg| {
-        arg.as_bytes_with_nul().as_ptr() as *const i8
+        arg.as_bytes_with_nul().as_ptr() as *const c_char
```

Comment:
Here we call `.as_bytes_with_nul().as_ptr() as *const c_char`. We should change this to a plain `.as_ptr()`.